GitUI/SpellChecker/WordExtractor.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Text;
+
+namespace GitUI.SpellChecker
+{
+    public static class WordExtractor
```

Comment:
The implementation looks good however now I am not 100% happy with the API itself.

* If the class only had `Extract` method the name would be suitable. But `WordExtractor.IsSeparator` does not look right. Perhaps `SpellCheckerHelper` would be more appropriate with `ExtractWord` (`GetWordAtCursor`) and `IsSeparator` methods in it.

* Personally I am not a fan of static classes and methods because they make testing stories difficult and preclude implementation substitution. I very much prefer the interface-driven development.
There are cases when static helper and extension classes are justified but too often developer abuse them.

* If you insist on keeping the class static then perhaps it is worth considering changing the methods to extensions - the following may read better:
```
if (TextBox.Text.Length > TextBox.SelectionStart && TextBox.Text[TextBox.SelectionStart].IsSeparator())
    return null;

return TextBox.Text.ExtractWord(TextBox.SelectionStart - 1);
```

I'm open for discussion on these.

Reply:
- I agree with you, the reason of `WordExtractor.IsSeparator` is because the last moment I decided reuse of `IsSeparator` method and didn't notice design issue.
- I thought about making `WordExtractor` a normal class, I have changed it's behavior to a static one because I didn't want to create it each time the character was typed. Of course I can create or even better to inject it in ctor, but the class is so large, that I feel a bit uncomfortable when I add a dependency in ctor and use it only at line 800. The other solution would be making a singleton class with `private ctor` and `public static readonly Instance`. Any way I think I should change it to a normal class.

Reply:
I have no problem, in fact I quite like dependency injection. This way we can slowly refactor the codebase - bit by bit. Maybe one day we will be in a position to plug in an IoC container in :)
You can instantiate the the extractor either before or in the ctor.